### PR TITLE
Move YAML, CRDT and memory classes to internal package

### PR DESCRIPTION
**Pull request by mmedenjak**
Another "broad stroke" change in moving private API from public to
internal package. Also fixed checkstyle by adding package.info and
adding annotations where needed. Other than moving the classes
(package/import changes) and checkstyle, there is no code or behaviour
change.

EE: https://github.com/hazelcast/hazelcast-enterprise/pull/3171

Classes moved:
com.hazelcast.config.yaml.ElementAdapter
com.hazelcast.config.yaml.EmptyNamedNodeMap
com.hazelcast.config.yaml.EmptyNodeList
com.hazelcast.config.yaml.NamedNodeMapAdapter
com.hazelcast.config.yaml.NodeListMappingAdapter
com.hazelcast.config.yaml.NodeListScalarAdapter
com.hazelcast.config.yaml.NodeListSequenceAdapter
com.hazelcast.config.yaml.ScalarTextNodeAdapter
com.hazelcast.config.yaml.SingletonNodeList
com.hazelcast.config.yaml.W3cDomUtil
com.hazelcast.config.yaml.YamlDomChecker
com.hazelcast.config.yaml.YamlOrderedMapping
com.hazelcast.config.yaml.YamlOrderedMappingImpl

com.hazelcast.crdt.pncounter.operations.AbstractPNCounterOperation
com.hazelcast.crdt.pncounter.operations.AddOperation
com.hazelcast.crdt.pncounter.operations.CRDTTimestampedLong
com.hazelcast.crdt.pncounter.operations.GetOperation
com.hazelcast.crdt.pncounter.PNCounterImpl
com.hazelcast.crdt.pncounter.PNCounterProxy
com.hazelcast.crdt.pncounter.PNCounterReplicationOperation
com.hazelcast.crdt.pncounter.PNCounterService

com.hazelcast.crdt.AbstractCRDTReplicationOperation
com.hazelcast.crdt.CRDT
com.hazelcast.crdt.CRDTDataSerializerHook
com.hazelcast.crdt.CRDTMigrationTask
com.hazelcast.crdt.CRDTReplicationAwareService
com.hazelcast.crdt.CRDTReplicationContainer
com.hazelcast.crdt.CRDTReplicationMigrationService
com.hazelcast.crdt.CRDTReplicationTask
com.hazelcast.crdt.ReplicatedVectorClocks

com.hazelcast.memory.DefaultGarbageCollectorStats
com.hazelcast.memory.DefaultMemoryStats
com.hazelcast.memory.GarbageCollectorStats
com.hazelcast.memory.GCStatsSupport
com.hazelcast.memory.MemoryStats
com.hazelcast.memory.MemoryStatsSupport

com.hazelcast.cache.hidensity.HiDensityCacheRecord

com.hazelcast.elastic.map.BehmMemoryBlockProcessor
com.hazelcast.elastic.map.BehmSlotAccessor
com.hazelcast.elastic.map.BehmSlotAccessorFactory
com.hazelcast.elastic.map.BinaryElasticHashMap
com.hazelcast.elastic.map.ConcurrentElasticHashMap
com.hazelcast.elastic.map.ElasticHashMap
com.hazelcast.elastic.map.NativeBehmSlotAccessorFactory
com.hazelcast.elastic.map.NativeMemoryDataAccessor
com.hazelcast.elastic.map.SampleableElasticHashMap

com.hazelcast.elastic.queue.AbstractBlockingElasticQueue
com.hazelcast.elastic.queue.AbstractElasticQueue
com.hazelcast.elastic.queue.ArrayBlockingElasticQueue
com.hazelcast.elastic.queue.ArrayElasticQueue
com.hazelcast.elastic.queue.BlockingElasticQueue
com.hazelcast.elastic.queue.ConcurrentLinkedElasticQueue
com.hazelcast.elastic.queue.LinkedBlockingElasticQueue
com.hazelcast.elastic.queue.LinkedElasticQueue

com.hazelcast.elastic.tree.BinaryElasticNestedTreeMap
com.hazelcast.elastic.tree.ComparableComparator
com.hazelcast.elastic.tree.MapEntryFactory
com.hazelcast.elastic.tree.NativeBinaryElasticNestedTreeMap

com.hazelcast.elastic.map.ConcurrentElasticMap
com.hazelcast.elastic.map.ElasticMap

com.hazelcast.elastic.queue.ElasticQueue
com.hazelcast.elastic.queue.LongArrayBlockingQueue
com.hazelcast.elastic.queue.LongArrayQueue
com.hazelcast.elastic.queue.LongBlockingQueue
com.hazelcast.elastic.queue.LongConcurrentLinkedQueue
com.hazelcast.elastic.queue.LongConsumer
com.hazelcast.elastic.queue.LongLinkedBlockingQueue
com.hazelcast.elastic.queue.LongLinkedQueue
com.hazelcast.elastic.queue.LongQueue

com.hazelcast.elastic.tree.impl.EntryIterator
com.hazelcast.elastic.tree.impl.EntryValuesIterator
com.hazelcast.elastic.tree.impl.RedBlackTreeNode
com.hazelcast.elastic.tree.impl.RedBlackTreeStore

com.hazelcast.elastic.tree.OffHeapComparator
com.hazelcast.elastic.tree.OffHeapTreeEntry
com.hazelcast.elastic.tree.OffHeapTreeStore
com.hazelcast.elastic.tree.OrderingDirection

com.hazelcast.elastic.CapacityUtil
com.hazelcast.elastic.LongArray
com.hazelcast.elastic.LongIterator
com.hazelcast.elastic.SlottableIterator

com.hazelcast.memory.AbstractPoolingMemoryManager
com.hazelcast.memory.AddressQueue
com.hazelcast.memory.DestroyedAddressQueue
com.hazelcast.memory.FreeMemoryChecker
com.hazelcast.memory.GarbageCollectable
com.hazelcast.memory.GarbageCollector
com.hazelcast.memory.GlobalPoolingMemoryManager
com.hazelcast.memory.HazelcastMemoryManager
com.hazelcast.memory.MemoryBlock
com.hazelcast.memory.MemoryBlockAccessor
com.hazelcast.memory.NativeMemoryStats
com.hazelcast.memory.package-info
com.hazelcast.memory.PooledNativeMemoryStats
com.hazelcast.memory.PoolingMemoryManager
com.hazelcast.memory.SimpleGarbageCollector
com.hazelcast.memory.StandardMemoryManager
com.hazelcast.memory.ThreadLocalPoolingMemoryManager

### CODE REVIEW

File: hazelcast-build-utils/src/test/resources/manifest.mf [hunk 1]
```diff
@@ -252,7 +252,7 @@ Export-Package: com.hazelcast.cluster.impl;uses:="com.hazelcast.loggin
  memory.impl,javax.management,javax.management.openmbean,sun.misc,com.
  hazelcast.spi.properties,com.hazelcast.client.impl,com.hazelcast.conf
  ig,com.hazelcast.spi.impl.executionservice,com.hazelcast.internal.clu
- ster.impl,com.hazelcast.cluster.memberselector,com.hazelcast.memory,c
+ ster.impl,com.hazelcast.cluster.memberselector,com.hazelcast.internal.memory,c
```
Comment (vojtechtoman) [inline hazelcast-build-utils/src/test/resources/manifest.mf]: This is probably irrelevant, but shouldn't the test file `manifest.mf` be either: updated to say `version="4.0.SNAPSHOT"`, or: kept intact to use the old package structure?

Comment (mmedenjak) [reply hazelcast-build-utils/src/test/resources/manifest.mf]: Good question. I would say updated but it's more of a wild guess as I have no real experience or knowledge with manifest.mf

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
Thank you for the review, guys!

**Pull request merged**